addons/binding/org.openhab.binding.innogysmarthome/src/main/java/org/openhab/binding/innogysmarthome/internal/InnogyWebSocket.java

```diff
@@ -0,0 +1,146 @@
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketError;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketMessage;
+import org.eclipse.jetty.websocket.api.annotations.WebSocket;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.openhab.binding.innogysmarthome.handler.InnogyBridgeHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link InnogyWebSocket} implements the websocket for receiving constant updates
+ * from the innogy SmartHome web service.
+ *
+ * @author Oliver Kuhl - Initial contribution
+ */
+@WebSocket
+public class InnogyWebSocket {
+
+    private Logger logger = LoggerFactory.getLogger(InnogyWebSocket.class);
+    private final CountDownLatch closeLatch;
+    private Session session;
+    private org.openhab.binding.innogysmarthome.internal.listener.EventListener eventListener;
+    private WebSocketClient client;
+    private final URI webSocketURI;
+    private final int maxIdleTimeout;
+
+    /**
+     * Constructs the {@link InnogyWebSocket}.
+     *
+     * @param bridgeHandler the responsible {@link InnogyBridgeHandler}
+     * @param webSocketURI the {@link URI} of the websocket endpoint
+     * @param maxIdleTimeout
+     */
+    public InnogyWebSocket(InnogyBridgeHandler bridgeHandler, URI webSocketURI, int maxIdleTimeout) {
+        this.eventListener = bridgeHandler;
+        this.closeLatch = new CountDownLatch(1);
+        this.webSocketURI = webSocketURI;
+        this.maxIdleTimeout = maxIdleTimeout;
+    }
+
+    /**
+     * Starts the {@link InnogyWebSocket}.
+     *
+     * @throws Exception
+     */
+    public synchronized void start() throws Exception {
+        SslContextFactory sslContextFactory = new SslContextFactory();
+        // sslContextFactory.setTrustAll(true); // The magic
+
+        if (client == null || client.isStopped()) {
+            client = new WebSocketClient(sslContextFactory);
+            client.setMaxIdleTimeout(this.maxIdleTimeout);
+            client.start();
+        }
+
+        if (session != null) {
+            session.close();
+        }
+
+        logger.debug("Connecting to innogy WebSocket...");
+        session = client.connect(this, webSocketURI).get();
+    }
+
+    /**
+     * Stops the {@link InnogyWebSocket}.
+     */
+    public synchronized void stop() {
+        if (isRunning()) {
+            logger.debug("Closing session...");
+            session.close();
+            session = null;
+        } else {
+            session = null;
+            logger.trace("Stopping websocket ignored - was not running.");
+        }
+    }
+
+    /**
+     * Return true, if the websocket is running.
+     *
+     * @return
+     */
+    public synchronized boolean isRunning() {
+        return session != null && session.isOpen();
+    }
+
+    @OnWebSocketConnect
+    public void onConnect(Session session) {
+        logger.info("Connected to innogy WebSocket.");
```

Comment:
As info level is a message for the end user rather than a developer, you might simply want to say "Connected to Innogy Webservice."